src/core/public/plugins/plugin.ts

```diff
@@ -118,6 +123,8 @@ export class PluginWrapper<
       throw new Error(`Plugin "${this.name}" can't be started since it isn't set up.`);
     }
 
+    this.startDependencies$.next([startContext, plugins]);
+
     return await this.instance.start(startContext, plugins);
```

Comment:
Maybe we don't care, but timing is important in some cases so, should `startDependencies$` really resolves before `this.instance.start` ?

Reply:
Good point, I think it's important that your plugin's `start` method is invoked before this is resolved so that any internal APIs/services your plugin needs are completely initialized correctly before this kicks off. Will add a test and update this.